### ISSUE TITLE
Use of pre created Entra ID SPN to deploy the resources

### ISSUE THREAD

**Issue by sepenet** [microsoft/finops-toolkit#879]
As a customer FinOps toolkit user I need the toolkit deployment pipeline to be able to use an already created SPN provided by Entra ID team as my user does not have permission to create app in Entra ID.

**Comment by flanakin** [comment]
@sepenet Is this for FinOps hubs managed exports?

**Comment by sepenet** [comment]
Hi @flanakin, it is to install the finops toolkit, but I guess it will be also valid request to setup the exports

**Comment by flanakin** [comment]
Thanks for the details. I added it to the backlog. I'll talk to @MSBrett to see what's needed for this. In the meantime, please create exports manually.

**Comment by MSBrett** [comment]
> Hi @flanakin, it is to install the finops toolkit, but I guess it will be also valid request to setup the exports

To confirm - you're having issues installing FTK because you don't have permissions to create the managed identities we use during deployment for managing the triggers, etc. and need to supply a pre-created SPN instead?

**Comment by sepenet** [comment]
Hello

Yes, CX contact has issue to install FTK, he does not have permission to create managed identities and security team would like him to use pre created SPN with limited permissions instead.
the same SPN should also be used to manage the exports

Thanks for your support
________________________________

**Comment by AErmie** [comment]
The customer I am assisting with setting up the FinOps Toolkit, is also interested in this feature. Using a pre-created Managed Identity in the target environment is especially important in regulated/government-based environments.

**Comment by flanakin** [comment]
@sepenet Would you mind giving this a 👍? We're tracking the issues with the most votes for prioritization and I'd like to make sure this gets everyone's votes 🙂

**Comment by AErmie** [comment]
Quick follow-up question that came up recently.

Since the Managed Identity for the Data Factory is used to interact with Azure Cost Management, exports, etc., could we add/attach a 
 **User assigned managed identity** to the Data Factory?

The scenario we're encountering is if we have to re-deploy the FTK, then we have to go back and ask the team that manages Entra ID to re-grant the new Managed Identity the proper EA Department Reader permissions.

Since it is not yet support to [use an existing Service Principal](https://github.com/microsoft/finops-toolkit/issues/879), we're hoping that if the Data Factory's **System Assigned** Managed Identity does not have the correct permissions at the assigned scope, that it would "fail back" to use the UAMI. 

This would allow us to still have a single Service Principal with the correct permissions, and not have to re-grant permissions anytime the FTK might have to be re-deployed.

**Comment by flanakin** [comment]
@AErmie I don't think it will fall back automatically. I think it'd have to be configured to use one or another. I'm not familiar with this but I'm assuming it's doable.

@MSBrett, you know more about this space. Any thoughts?
